### DISCUSSION TITLE
SO Ingest Integrations - Azure AD Logs

### DISCUSSION THREAD

**Discussion by Git-Me-Some-Hub** [Security-Onion-Solutions/securityonion#4170] [Unsupported Versions] [answered]
I am looking to evaluate SO for a more enterprise deployment, but I guess I am not sure how this works with integration from say Azure AD login when using Office 365?  I see that Elastic Search has those integration modules, but are those able to be installed on SO, or is it completely different and I will have to figure out my own ingestion parameters? 

Thanks.

**Comment by Git-Me-Some-Hub** [comment]
I was hoping that I could have gotten an answer fairly easy on this question? I see that Elastic has these type of modules in the x-pack, but I am just not sure if x-pack is something I can install for Security Onion, or get Office 365/Azure login logs.

**Comment by weslambert** [comment, marked as answer]
Filebeat provides an Azure and O365 module that can be leveraged.  You can do this today, but that configuration will need to change in the future, where we plan to make it a lot easier to use these modules and their use will be orchestrated using Salt.  

Manually enabling these modules at the moment requires:

- enabling via modules.yml
- loading associated ingest pipeline(s)
- routing to ingest pipeline(s) via Logstash Elasticsearch output

We won't go over this configuration since we will be changing the way that they are enabled/used, but this should give you a high-level overview of what's required.

**Reply by Git-Me-Some-Hub** [reply]
Oh great, thanks. I will start looking how to accomplish this.

**Reply by thegreekman** [reply]
Hey, did you ever figure out how to ingest logs from azure/office 365? If so, do you think you can direct me in the right direction?

**Reply by Git-Me-Some-Hub** [reply]
Yup I did. Two major things you have to have:

1. Be on SO 2.60 and above. This is the version that they had the Modules integration setup in a more use way. If you are behind that there is still away, but you just have to have an external source with Filebeats installed, which should come with the modules. 

- You will also need to setup the Module configurations with the API site, App ID, and API Secret. [Filebeats Module Config](https://www.elastic.co/guide/en/beats/filebeat/7.13/filebeat-module-o365.html)

2. The other piece is probably the harder part, it is setting up your own Enterprise App that has access to the O365 Tenant logs and can allow the module to reach out and get them. 

-  There are two routes, EventHub which cost money because you have to spin up a Azure VM to handle the Data. I think you might get access to more data this route. Not sure.
- The second route, Free and what I used was 365 Management API, this is the older guide I used from Microsoft. Note I did have to do some trial and error because they have changed the UI, and I was not sure on everything. [O365 Management API Guide](https://docs.microsoft.com/en-us/office/office-365-management-api/get-started-with-office-365-management-apis)


Let me know if you need any further instruction. Unless you know more than I about Enterprise App setup, that is where you might have the trouble. Ping me and I can dig into some specific questions.
